packages/core/src/types.ts

```diff
@@ -388,6 +388,8 @@ export interface InvokeDefinition<
   TEvent extends EventObject
 > {
   id: string;
```

Comment:
```suggestion
  id: string | undefined;
```

If I'm reading [the code in this PR](https://github.com/statelyai/xstate/pull/3837/files#diff-fc91146eff717b2bcd6adbc0955ca0549346ad7bef9aa2278e0a75a9793a66b9R147-R148) right looks like `id` is optional now? Looks like if it's not defined it'll be set to the returned ID from registering the actor with the system.

Reply:
The `id` is "local" to the actor, so it can reference its child by `id`. The `key` makes it discoverable throughout the system.

The `id` is still necessary and autogenerated if not specified, and different from the `key`.

<img width="714" alt="CleanShot 2023-03-04 at 14 10 01@2x" src="https://user-images.githubusercontent.com/1093738/222930845-43a0b912-0b18-49c8-a07f-78ebcb4c1fec.png">

Reply:
Right, I get that. It just seems like passing an `id` is optional, since it'll be based off the `sessionId` returned from registering with the system if not specified. I was wondering about it because the example in the changeset didn't have an `id` and I thought that was required for all invoked actors.

Reply:
Ah okay I get what you're saying. The `id` is indeed optional; this is the `InvokeDefinition` type used internally with the resolved ID (either the given one or a generated one), which is always provided

Reply:
Hah, oops. What I get for trying to find type definitions in the diff view 😅